src/library/scala/collection/mutable/BufferLike.scala

```diff
@@ -105,15 +105,18 @@ trait BufferLike[A, +This <: BufferLike[A, This] with Buffer[A]]
     */
   def remove(n: Int): A
 
-  /** Removes a number of elements from a given index position.
+  /** Removes a number of elements from a given index position.  Subclasses of `BufferLike`
+   *  will typically override this method to provide better performance than `count`
+   *  successive calls to single-element `remove`.
    *
    *  @param n  the index which refers to the first element to remove.
    *  @param count  the number of elements to remove.
    *  @throws   IndexOutOfBoundsException if the index `n` is not in the valid range
-   *            `0 <= n <= length - count`.
+   *            `0 <= n <= length - count` (with `count > 0`).
```

Comment:
The rationale here is that it isn't an `IndexOutOfBoundsException` if you're not actually trying to remove anything--you aren't doing any indexing.  `remove(n, 0)` should be a no-op for every collection since it asks you to do nothing.  `remove(n, -1)` logically means to add an element, so that's an `IllegalArgumentException`.